test/clj/rems/api/test_permissions.clj

```diff
@@ -8,22 +8,68 @@
   :once
   api-fixture)
 
+(defn- validate-alice-result [data]
+  (is (= 1 (count data)))
+  (is (contains? data :ga4gh_visa_v1))
+  (is (vector? (:ga4gh_visa_v1 data)))
+  (is (string? (first (:ga4gh_visa_v1 data)))))
+
 (deftest permissions-test
-  (testing "listing without authentication"
-    (let [response (-> (request :get (str "/api/permissions/userx"))
-                       handler)
-          body (read-body response)]
-      (is (= "unauthorized" body))))
-
-  (testing "all"
-    (let [api-key "42"
-          data (-> (request :get "/api/permissions/alice")
-                   (authenticate api-key "developer")
-                   handler
-                   read-ok-body)]
-      (is (= 1 (count data)))
-      (is (contains? data :ga4gh_visa_v1))
-      (is (vector? (:ga4gh_visa_v1 data)))
-      (is (string? (first (:ga4gh_visa_v1 data)))))))
+  (let [api-key "42"]
+        (testing "listing without authentication"
+          (let [response (-> (request :get (str "/api/permissions/userx"))
+                             handler)
+                body (read-body response)]
+            (is (= "unauthorized" body))))
+
+        (testing "listing without appropriate role"
+          (let [response (-> (request :get (str "/api/permissions/alice"))
+                             (authenticate "42" "approver1")
+                             handler)
+                body (read-body response)]
+            (is (= "unauthorized" body))))
+
+        (testing "all for alice as malice"
+          (let [response (-> (request :get (str "/api/permissions/alice"))
+                         (authenticate "42" "malice")
+                         handler)
+                body (read-body response)]
+            (is (= "unauthorized" body))))
+
+        (testing "all for alice as alice"
+          (let [data (-> (request :get (str "/api/permissions/alice"))
+                         (authenticate "42" "alice")
```

Comment:
this test doesn't catch the bug I mention above because there is only one entitlement. perhaps there should be two entitlements in the db so that we'll see that alice sees only her own entitlements?